src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.IncrementalMemberEditAnalyzer.cs

```diff
@@ -126,59 +127,59 @@ public async Task<ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<Diagnos
                     ImmutableArray<TextSpan> oldMemberSpans,
-                    PooledDictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
+                    Dictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
                 {
-                    if (analyzersWithState.Count == 0)
+                    if (analyzers.Count == 0)
                         return;
 
                     executor = executor.With(analysisScope.WithSpan(changedMember.FullSpan));
-                    await ExecuteAnalyzersAsync(executor, analyzersWithState, oldMemberSpans, builder).ConfigureAwait(false);
+                    await ExecuteAnalyzersAsync(executor, analyzers, oldMemberSpans, builder).ConfigureAwait(false);
                 }
 
                 async Task ExecuteDocumentBasedAnalyzersAsync(
-                    ArrayBuilder<StateSet> analyzersWithState,
+                    ArrayBuilder<DiagnosticAnalyzer> analyzers,
                     ImmutableArray<TextSpan> oldMemberSpans,
-                    PooledDictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
+                    Dictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
                 {
-                    if (analyzersWithState.Count == 0)
+                    if (analyzers.Count == 0)
                         return;
 
                     executor = executor.With(analysisScope.WithSpan(null));
-                    await ExecuteAnalyzersAsync(executor, analyzersWithState, oldMemberSpans, builder).ConfigureAwait(false);
+                    await ExecuteAnalyzersAsync(executor, analyzers, oldMemberSpans, builder).ConfigureAwait(false);
                 }
 
                 async Task ExecuteAnalyzersAsync(
                     DocumentAnalysisExecutor executor,
-                    ArrayBuilder<StateSet> analyzersWithState,
+                    ArrayBuilder<DiagnosticAnalyzer> analyzers,
                     ImmutableArray<TextSpan> oldMemberSpans,
-                    PooledDictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
+                    Dictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> builder)
                 {
                     var analysisScope = executor.AnalysisScope;
 
                     Debug.Assert(changedMember != null);
                     Debug.Assert(analysisScope.Kind == AnalysisKind.Semantic);
 
-                    foreach (var analyzerWithState in analyzersWithState)
+                    foreach (var analyzer in analyzers)
                     {
-                        var diagnostics = await computeAnalyzerDiagnosticsAsync(analyzerWithState.Analyzer, executor, cancellationToken).ConfigureAwait(false);
-                        builder.Add(analyzerWithState.Analyzer, diagnostics);
+                        var diagnostics = await computeAnalyzerDiagnosticsAsync(analyzer, executor, cancellationToken).ConfigureAwait(false);
```

Comment:
Was trying to look - does this (and the other func) actually need to be passed in as a function at this point?  This was the only caller I could find - https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs#L299

Reply:
good question.  will change if so.

Reply:
c59a123b668